R/drive_search.R

```diff
@@ -53,11 +53,13 @@ drive_search <- function(pattern = NULL, type = NULL, ..., verbose = TRUE) {
   }
 
   if (!is.null(type)) {
-    mime_type <- drive_mime_type(type, verbose = verbose)
-    if (!is.null(mime_type)) {
-    params$q <- glue::collapse(c(params$q, paste0("mimeType = '", mime_type,"'")),
-                               sep = " and ")
-    }
+    mime_type <- drive_mime_type(type)
+    ## if they are all NA, this will error, because drive_mime_type
+    ## doesn't allow it, but if one is NA and the others aren't,
+    ## this will still pass the ok ones through.
+    params$q <- paste(c(params$q,
+                        paste0("mimeType = '", mime_type,"'", collapse = " or ")),
```

Comment:
I've left this (I'm about to push), but did tweak so we drop the `NA`s before forming the `q` clause. Maybe this doesn't hurt anything: `"mimeType = 'text/csv' or mimeType = 'NA' or mimeType = 'NA' or mimeType = 'application/pdf'"` but it seems odd.